aten/src/ATen/core/type.cpp

```diff
@@ -585,6 +585,22 @@ bool is_null_or_equal(c10::optional<T> a, c10::IntArrayRef b) {
   return !a.has_value() || a.value() == b;
 }
 
+
+
+ bool matchStrides(TensorType* ttp, const at::Tensor& t) {
```

Comment:
Maybe make matchStrides static, and can `TensorType* ttp` be `const TensorType& tt`?

Reply:
ah, yes, a great point!